### PR TITLE
Upgrade to Vault 0.7.0

### LINKED ISSUE #13: Upgrade Vault to latest version
The Current version is 0.7.0.

Thanks

### PR DESCRIPTION
**Pull request by jwhitcraft**
## The Problem:
Vault is not at the latest version, fixes #13
## The Fix:
Update the Dockerfile and the deployment yaml files for vault

### CODE REVIEW

**Comment by jwhitcraft** [comment]
@rfay I've got this running on my test cluster, and it works good.  I had to add the redirect_url as it's as complaining that it wasn't set.

**Comment by rfay** [comment]
Thanks - This looks to be uneventful. I may not be able to test it in the short term, but will commit if others test and report back.

File: deployments/vault-1.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
         fsGroup: 1000
       containers:
         - name: vault
-          image: "drud/vault:0.6.3-2"
+          image: "drud/vault:0.7.0-1"
```
Comment (rfay) [inline deployments/vault-1.yaml]: You must be testing with something other than this :) https://hub.docker.com/r/drud/vault/tags/

**Comment by jwhitcraft** [comment]
@rfay yes I was testing with an image on my personal registry and i left the -1 in there.  I've removed them.  Clearly it won't work till you push the image up.

**Comment by rfay** [comment]
I went ahead and built and pushed the drud/vault:0.7.0 image, so this should be testable now. Thanks!

**Comment by jwhitcraft** [comment]
@rfay i've been running it over night in my test env and no issues so far.

**Comment by rfay** [comment]
I've deployed this on our staging vault without incident. @beeradb if you could take a look at that and see if you have any concerns, I think this is fine to merge.

Note that anybody upgrading needs to read the [vault CHANGELOG](https://github.com/hashicorp/vault/blob/master/CHANGELOG.md) reading 0.6.4 through 0.7.0. 

In the case of drud, there's a change to the use of "list" in policies:
>List Operations Always Use Trailing Slash: Any list operation, whether via the GET or LIST HTTP verb, will now internally canonicalize the path to have a trailing slash. This makes policy writing more predictable, as it means clients will no longer work or fail based on which client they're using or which HTTP verb they're using. However, it also means that policies allowing list capability must be carefully checked to ensure that they contain a trailing slash; some policies may need to be split into multiple stanzas to accommodate.

**Comment by beeradb** [comment]
Tested this PR this morning and it all looks good to me. 👍

**Comment by rfay** [comment]
Thanks so much for this @jwhitcraft ! If you want to give consul 8.1 a try it would be lovely too :) https://github.com/drud/vault-consul-on-kube/issues/12

**Comment by jwhitcraft** [comment]
@rfay was going to look at it later today :P

**Pull request merged**